frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx

```diff
@@ -0,0 +1,105 @@
+import { Route } from "react-router";
+
+import { setupEnterprisePlugins } from "__support__/enterprise";
+import { mockSettings } from "__support__/settings";
+import { renderWithProviders } from "__support__/ui";
+import type { EmbedResource } from "metabase/public/lib/types";
+import type { TokenFeatures } from "metabase-types/api";
+import { createMockUser } from "metabase-types/api/mocks";
+
+import type { EmbedModalContentProps } from "../EmbedModalContent";
+import { EmbedModalContent } from "../EmbedModalContent";
+
+type EnableEmbedding = {
+  static?: boolean;
+  interactive?: boolean;
+  sdk?: boolean;
+};
+
+export interface SetupOpts {
+  enableEmbedding?: EnableEmbedding;
+  props?: Partial<EmbedModalContentProps>;
+  hasEnterprisePlugins?: boolean;
+  tokenFeatures?: TokenFeatures;
+}
+
+export function setup(
+  {
+    enableEmbedding: {
+      static: enableEmbeddingStatic = false,
+      interactive: enableEmbeddingInteractive = false,
+      sdk: enableEmbeddingSdk = false,
+    } = {
+      static: false,
+      interactive: false,
+      sdk: false,
+    },
+    props: {
+      embedType = null,
+      resource = {} as EmbedResource,
+      resourceType = "dashboard",
+      resourceParameters = [],
+      goToNextStep = jest.fn(),
+      getPublicUrl = jest.fn(_resource => "some URL"),
+      onUpdateEmbeddingParams = jest.fn(),
+      onUpdateEnableEmbedding = jest.fn(),
+      onCreatePublicLink = jest.fn(),
+      onDeletePublicLink = jest.fn(),
+    } = {},
+    hasEnterprisePlugins,
+    tokenFeatures,
+  }: SetupOpts = {
+    props: {},
+  },
+) {
+  const settings = mockSettings({
+    "token-features": tokenFeatures,
+    "enable-embedding-static": enableEmbeddingStatic,
+    "enable-embedding-interactive": enableEmbeddingInteractive,
+    "enable-embedding-sdk": enableEmbeddingSdk,
+    "embedding-secret-key": "my_super_secret_key",
+  });
```

Comment:
The change here is to use `mockSetting` and set the value in the `storeInitialState` because our test relies on 2 things.

`MetabaseSetting.get` is achieved by calling `mockSettings`, and if our code read the token features from the state, we need to also mock these token features in the store.